### PR TITLE
[SW-2305] Ensure that call /3/Shutdown handles H2OContext stop in case of Sparkling Water ( via Flow Proxy)

### PR DESCRIPTION
**Pull request by jakubhava**
This is to allow Steam to properly stop Sparkling Water when calling shutdown endpoint

### CODE REVIEW

File: core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      case invalid => throw new H2ONotFoundArgumentException(s"Invalid endpoint $invalid")
+    }
+    sendResult(obj, resp)
+  }
+
+}
+
+object ShutdownServlet extends ServletRegister {
+
+  override protected def getEndpoints(): Array[String] = {
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala:42]: Shouldn't `getEndpoints()`  return ip addresses?

Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala:42]: Nope, we return the request path. The `getEndpoints` method should be renamed

**Comment by jakubhava** [comment]
The motivation is mentioned in the description above. Basically, we want to allow steam to stop H2O & Spark stack with the resulting YARN app having status as SUCCESS. Currently, Steam is killing the app and the app has the status KILLED

File: core/src/main/scala/ai/h2o/sparkling/backend/api/rdds/RDDsServlet.scala [hunk 2]
```diff
@@ -105,7 +105,7 @@ private[api] class RDDsServlet extends ServletBase {
 
 object RDDsServlet extends ServletRegister {
 
-  override protected def getEndpoints(): Array[String] = {
+  override protected def getRequestPaths(): Array[String] = {
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/api/rdds/RDDsServlet.scala:108]: nit: could be in one line

File: core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      case invalid => throw new H2ONotFoundArgumentException(s"Invalid endpoint $invalid")
+    }
+    sendResult(obj, resp)
+  }
+
+}
+
+object ShutdownServlet extends ServletRegister {
+
+  override protected def getRequestPaths(): Array[String] = {
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala:42]: nit: Could be in one line?

File: core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  override def doPost(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    val obj = req.getPathInfo match {
+      case null =>
+        H2OContext.get().foreach(_.stop(stopSparkContext = true))
+      case invalid => throw new H2ONotFoundArgumentException(s"Invalid endpoint $invalid")
+    }
+    sendResult(obj, resp)
+  }
+
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala:37]: nit: extra line

File: core/src/main/scala/ai/h2o/sparkling/backend/api/rdds/RDDsServlet.scala (hunk 2 above)
Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/api/rdds/RDDsServlet.scala]: 👍

**Pull request merged**